dm/dm/config/subtask.go

```diff
@@ -394,10 +395,29 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		c.MetaSchema = defaultMetaSchema
 	}
 
-	dirSuffix := "." + c.Name
-	if !strings.HasSuffix(c.LoaderConfig.Dir, dirSuffix) { // check to support multiple times calling
-		// if not ends with the task name, we append the task name to the tail
-		c.LoaderConfig.Dir += dirSuffix
+	// adjust dir
+	if c.Mode == ModeAll || c.Mode == ModeFull {
+		isS3, err := exstorage.IsS3Path(c.LoaderConfig.Dir)
+		if err != nil {
+			return terror.ErrConfigLoaderDirInvalid.Delegate(err, c.LoaderConfig.Dir)
+		}
+		if isS3 {
+			// check
+			if c.ImportMode != LoadModeSQL {
+				return terror.ErrConfigLoaderS3NotSupport.Generate(c.LoaderConfig.Dir)
+			}
+			newDir, err := exstorage.AdjustS3Path(c.LoaderConfig.Dir, c.Name+"."+c.SourceID)
```

Comment:
if user try to use dmctl get-config he/she will get a different config file from input ,Is this as expected?

Reply:
we will add suffix(taskname) before, and now s3 is like before, add suffix(taskname.sourceid), so i think this should still as expected.

Reply:
yes, this way can move storage type check into `ExternalStorage`. i think i will `dm/pkg/exstorage/utils.go`, because we need avoid repeated add suffix.